smartdashboard/views.py

```diff
@@ -341,3 +345,467 @@ def __init__(
         self.app_view = app_view
         self.ens_view = ens_view
         self.orc_view = orc_view
+
+    def update(self) -> None:
+        """Update all views within the OverviewView"""
+        self.exp_view.update()
+        self.app_view.update()
+        self.ens_view.update()
+        self.orc_view.update()
+
+
+@dataclass(frozen=True)
+class Files:
+    graph_file: str
+    table_file: str
+
+    def is_valid(self) -> bool:
+        """Check that paths are not empty strings and exist"""
+        if any(file_path == "" for file_path in (self.graph_file, self.table_file)):
+            return False
+
+        if not all(
+            os.path.exists(file_path)
+            for file_path in (self.graph_file, self.table_file)
+        ):
+            return False
+
+        return True
+
+
+class DualView(ViewBase):
+    """Base class for Telemetry Views that have table and chart elements"""
+
+    def __init__(
+        self,
+        shard: t.Optional[Shard],
+        table_element: DeltaGenerator,
+        graph_element: DeltaGenerator,
+        export_button: DeltaGenerator,
+    ):
+        self.shard = shard
+        self.table_element = table_element
+        self.graph_element = graph_element
+        self.export_button = export_button
+        self.window_size = 10000
+        self.telemetry_df = pd.DataFrame(columns=self.columns)
+        self.sampling = False
+        self.chart: t.Optional[alt.Chart] = None
+
+        if self.telemetry:
+            self.telemetry_df = self.set_telemetry_df()
```

Comment:
Not wrong, but tends to be a bit of a code smell in my book: We are setting an instance attr to the return value of an instance method. Can `self.telemetry_df` be an `@property` instead?

NOTE: no need to change this if this is done for performance/caching reasons, but if you could add a comment or something here explaining that that would be appreciated!

Reply:
It can't be a property because it's changing whenever there are additions to the graph dataframes. But I could set it like this:
```python
if self.telemetry:
    self.telemetry_df = self._load_data_update(skiprows=0)
```